src/openrct2/peep/Peep.cpp

```diff
@@ -3243,11 +3233,12 @@ int32_t peep_compare(const uint16_t sprite_index_a, const uint16_t sprite_index_
     uint8_t args[32]{};
 
     char nameA[256]{};
-    peep_a->FormatNameTo(args);
+    Formatter formatter(args);
+    peep_a->FormatNameTo(formatter);
     format_string(nameA, sizeof(nameA), STR_STRINGID, args);
 
     char nameB[256]{};
-    peep_b->FormatNameTo(args);
+    peep_b->FormatNameTo(formatter);
```

Comment:
```suggestion
    formatter = Formatter(args);
    peep_b->FormatNameTo(formatter);
```
Pretty sure you need to restart the formatter, otherwise you'll be appending things to where the previous one ended.
If you note, the `FormatNameTo(void* args)` starts the `Formatter` anew for every call.